src/main/java/de/tfelix/namegen/model/Transition.java

```diff
@@ -0,0 +1,152 @@
+        // Having a tree allows deterministic traversal
+	    runtimeTransition.distribution = new TreeMap<Character, Float>();
+        float observationRange = 1.0f;
+	    if(priorProbability >= Math.ulp(1.0)) {
+            // Prior is desired; initialise the alphabet.
+            UnicodeSet alphabet = LocaleData.getExemplarSet(this.locale, ES_STANDARD);
+            /* Observations need to be scaled so that the probability across the alphabet sums to 1.0 */
+            observationRange = (1.0f - priorProbability * alphabet.size());
+            if (observationRange < 0.0) {
+                logger.warn("The prior probability was meant to be the chance that any available letter would occur. " +
+                        "By specifying a probability of {} with {} letters in the alphabet means that there's no " +
+                        "room in the probability distribution to adjust for the letters that are more likely.",
+                        priorProbability, alphabet.size());
+                // Set a default prior − the show must go on
+                priorProbability = 1.0f / (2.0f*alphabet.size());
+                observationRange = 0.5f;  // Half of our outputs will be influenced by the observations
+            }
+            Iterator<String> iterator = alphabet.iterator();
+            while(iterator.hasNext()) {
+                Character letter = iterator.next().charAt(0);
+                runtimeTransition.distribution.put(letter, priorProbability);
+            }
+        }
+        for(Entry<Character, Integer> entry: observedChars.entrySet()) {
+            // Some observed characters (such as the ending token, hopefully) might not belong to the alphabet
+            float probability = observationRange * entry.getValue() / (float)(this.observations);
+            if (runtimeTransition.distribution.containsKey(entry.getKey())) {
+                Float prior = runtimeTransition.distribution.get(entry.getKey());
+                probability += prior;
+            }
+            runtimeTransition.distribution.put(entry.getKey(), probability);
+        }
+        return runtimeTransition;
+    }
+
+	/**
+	 * Deterministically pick a new character from this transition's probability distribution.
+	 * 
+	 * @param position
+	 *            A position in the probability distribution ∈ [0, 1.0].
+	 * @return A randomly picked character.
+	 */
+	public char pick(float position) throws RuntimeException {
+		if (position < 0 || position > 1.0) {
+			throw new IllegalArgumentException(String.format("Probability %g must be between 0 and 1.0", position));
+		}
+		if(distribution == null) {
+		    throw new RuntimeException("A transition was called for sampling before it had been built.");
+        }
+		float cumulation = 0f;
+		for (Entry<Character, Float> entry : distribution.entrySet()) {
+			/** By iterating and cumulating, it's easier to query specific letters and see their probability.
```

Comment:
yeah that would be a nice addition in case someone really want to pull out names :)